report/sql_tasks/tasks/report/create_from_scratch/04_hubspot_write/01_company_property_update/01_create_view.sql

```diff
@@ -0,0 +1,76 @@
+DROP VIEW IF EXISTS hubspot.company_property_update CASCADE;
+
+-- This view recreates the properties that we update on Hubspot companies
+CREATE VIEW hubspot.company_property_update AS (
+    WITH
+        metrics AS (
+            SELECT
+                column1 AS metric_name,
+                column2 AS role,
+                column3 AS timescale,
+                column4 AS offset
+            FROM (
+                VALUES
+                    ('lms_teachers_last_semester', 'teacher', 'semester', 1),
+                    ('lms_teachers_this_semester', 'teacher', 'semester', 0),
+                    ('lms_users_last_academic_year', 'user', 'academic_year', 1),
+                    ('lms_users_this_academic_year', 'user', 'academic_year', 0),
+                    ('lms_users_last_semester', 'user', 'semester', 1),
+                    ('lms_users_this_semester', 'user', 'semester', 0),
+                    ('lms_users_all_time', 'user', 'all_time', 0)
```

Comment:
Each of the metrics can be expressed naturally as a role, timescale and offset from the latest period.